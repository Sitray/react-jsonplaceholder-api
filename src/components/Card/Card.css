* {
  padding: 0;
  margin: 0;
}

body {
  background: #fefefe;
  font-family: sans-serif;
}

hr{
  border: 0;
  height: 1px;
  background: black;
}

.heading {
  text-align: center;
  font-size: 30px;
  margin-bottom: 50px;
}

.row {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  flex-flow: wrap;
 
}

.card {
  width: 20%;
  background: #fff;
  border: 1px solid #ccc;
  margin-bottom: 50px;
  transition: 0.3s;
  border-radius: 10px;
  font-size: 10px;
  width: 90%;
  animation: fadein;
  animation-duration: 10s;
  animation-delay: 10s;
}

.card-header {
  text-align: center;
  padding: 40px 15px;
  background: linear-gradient(to right, #4C5B61, #829191);
  color: #fff;
  border-radius: 10px;
}

.card-body {
  padding: 30px 20px;
  text-align: center;
  font-size: 18px;
}

.card-body .btn {
  display: block;
  color: #fff;
  text-align: center;
  background: #2C423F;
  margin-top: 30px;
  text-decoration: none;
  padding: 10px 5px;
  border-radius: 8px;
  border: none;
  cursor: pointer;
}

.btn:hover {
  background: #829191;
  box-shadow: 0 0 40px -10px rgba(0, 0, 0, 0.25);

}

.button-container{
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.card:hover {
  transform: scale(1.05);
  box-shadow: 0 0 40px -10px rgba(0, 0, 0, 0.25);
}

.textarea{
  resize: none;
  border: none;
  font-family: sans-serif;
  height: auto;
  text-overflow: ellipsis;
}

.textarea-open{
  resize: none;
  font-family: sans-serif;
  height: auto;
  text-overflow: ellipsis;
}

@media screen and (max-width: 1000px) {
  .card {
    width: 40%;
  }
}

@media screen and (max-width: 620px) {
  .container {
    width: 100%;
  }

  .heading {
    padding: 20px;
    font-size: 20px;
  }

  .card {
    width: 80%;
  }
}
