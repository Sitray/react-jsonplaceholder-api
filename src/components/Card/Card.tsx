import React from 'react';

import './Card.css';

interface Props {
     userId: number,
     title: string,
     body: string,

}

const Card = ({ userId, title, body }: Props) => (
  <div className="card">
    <div className="card-header">
      <h1>{userId}</h1>
      <h1>{title}</h1>
    </div>
    <div className="card-body">
      <p>
        {body}
      </p>
      <button className="btn" type="button">Read more</button>
    </div>
  </div>
);

export default Card;
