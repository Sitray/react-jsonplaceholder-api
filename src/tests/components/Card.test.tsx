/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Card from '../../components/Card/Card';

describe('Test on <Card />', () => {
  const initialState = { id: 1 };

  const mockStore = configureStore();

  const body = 'This is a test';
  const id = 1;
  const userId = 1;
  const title = 'Test';

  const wrapper = shallow(
    <Provider store={mockStore(initialState)}>
      <Card body={body} id={id} userId={userId} title={title} />
    </Provider>,
  );

  test('should render correctly with props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
