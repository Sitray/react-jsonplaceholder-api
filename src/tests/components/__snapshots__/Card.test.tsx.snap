// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Card /> should render correctly with props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Card
    body="This is a test"
    id={1}
    title="Test"
    userId={1}
  />
</ContextProvider>
`;
